Gems/EMotionFX/Code/MCore/Source/Algorithms.h

```diff
@@ -62,6 +64,13 @@ namespace MCore
     float MCORE_API DistanceToEdge(const AZ::Vector2& edgePointA, const AZ::Vector2& edgePointB, const AZ::Vector2& testPoint);
     AZ::Vector2 MCORE_API ClosestPointToPoly(const AZ::Vector2* polyPoints, size_t numPoints, const AZ::Vector2& testPoint);
 
+    // @param data The original data that we want to apply moving-average smooth algorithm on
+    // @param sampleNum The sample number we will take from each data point.
+    // When sampleNum = 1, we are taking 1 sample before the data point, and 1 sample after the data point, 3 samples point in total (include the original data point).
+    // And calculate the average.
+    // When sampleNum = 2, we are taking 5 samples point in total and take the average (2 before, 1 original and 2 after).
+    void MCORE_API MovingAverageSmooth(AZStd::vector<AZ::Vector3>& data, size_t sampleNum);
+    void MCORE_API MovingAverageSmooth(AZStd::vector<MCore::Compressed16BitQuaternion>& data, size_t sampleNum);
```

Comment:
Nit: I'd rename sampleNum into samplesInWindow or kernelSize or something. And I'd find it more logical if you can specify the 5 there (from the example in the comment) and you calc the 2 based on that inside the function.

Reply:
When kernel size is even number should we default to the closest odd number? Would the user be curious if kernel size 4 and 3 yield the same result?

Reply:
Hm... good question. @greerdv do you have any references here on what is a common practice?